lookout/style/typos/benchmarks/typo_commits_report.py

```diff
@@ -0,0 +1,193 @@
+"""Facilities to report the quality of a given model on a given dataset."""
+import csv
+import json
+import logging
+import os
+from typing import Any, Dict, Iterable, Iterator, List, Optional, Sequence, Tuple
+
+
+from lookout.core.analyzer import ReferencePointer
+from lookout.core.api.service_analyzer_pb2 import Comment
+from lookout.core.api.service_data_pb2 import Change, File
+from lookout.core.data_requests import DataService, request_files
+import pandas
+
+from lookout.style.cloner import Cloner
+from lookout.style.format.benchmarks.quality_report import handle_input_arg
+from lookout.style.format.utils import generate_comment
+from lookout.style.reporter import Reporter
+from lookout.style.typos import IdTyposAnalyzer
+from lookout.style.typos.analyzer import TypoFix
+
+
+class TyposAnalyzerSpy(IdTyposAnalyzer):
+    """
+    The Analyzer which returns fixes found by IdTyposAnalyzer for all file content as JSON \
+    structures.
+    """
+
+    def run(self, ptr: ReferencePointer, data_service: DataService) -> Iterable[TypoFix]:
+        """
+        Run `generate_typos_fixes` for all lines and all files in ptr_from revision.
+
+        :param ptr: Git repository state pointer to the revision that should be analyzed.
+        :param data_service: Connection to the Lookout data retrieval service to get the files.
+        :return: Generator of fixes for each file.
+        """
+        files = request_files(data_service.get_data(), ptr,
+                              contents=True, uast=True, unicode=False)
+        return self.generate_typos_fixes([
+            Change(base=f, head=File(path=f.path, language=f.language)) for f in files])
+
+    def analyze(self, ptr_from: ReferencePointer, ptr_to: ReferencePointer,
+                data_service: DataService, **data) -> List[Comment]:
+        """
+        Return the list of `TypoFix`-es as Comments.
+
+        `TypoFix`-es are generated with `run()` method.
+
+        :param ptr_from: The Git revision to analyze.
+        :param ptr_to: Not used. ptr_from is used for both model training and analysis.
+        :param data_service: The channel to the data service in Lookout server to query for \
+                             UASTs, file contents, etc.
+        :param data: Extra data passed into the method. Used by the decorators to simplify \
+                     the data retrieval.
+        :return: List of Typo Fixes information in a JSON format required for further analysis.
```

Comment:
```suggestion
        :return: List of typo fixes in JSON.
```

Reply:
btw: what JSON? we return Python objects here

Reply:
JSON inside `Comment.text`. Will clarify.